src/components/CalendarContainer.ts

```diff
@@ -0,0 +1,526 @@
+import { Component, ReactChild, createElement } from "react";
+
+import { Calendar, CalendarEvent, CustomViews, View } from "./Calendar";
+
+interface WrapperProps {
+    friendlyId: string;
+    mxform: mxui.lib.form._FormBase;
+    mxObject: mendix.lib.MxObject;
+    readOnly: boolean;
+    style: string;
+}
+
+export interface CalendarContainerProps extends WrapperProps {
+    titleAttribute: string;
+    startAttribute: string;
+    endAttribute: string;
+    allDayAttribute: string;
+    eventColor: string;
+    titleAttributeContext: string;
+    startAttributeContext: string;
+    endAttributeContext: string;
+    allDayAttributeContext: string;
+    eventColorContext: string;
+    defaultView: View;
+    startPositionAttribute: string;
+    dataSource: DataSource;
+    eventEntity: string;
+    entityConstraint: string;
+    dataSourceMicroflow: string;
+    dataSourceNanoflow: Nanoflow;
+    popup: boolean;
+    selectable: boolean;
+    editable: "default" | "never";
+    onClickEvent: OnClickEventOptions;
+    onClickSlotEvent: OnClickEventOptions;
+    eventMicroflow: string;
+    eventNanoflow: Nanoflow;
+    slotMicroflow: string;
+    slotNanoflow: Nanoflow;
+    onDropEvent: OnClickEventOptions;
+    onDropMicroflow: string;
+    onDropNanoflow: Nanoflow;
+    onResizeEvent: OnClickEventOptions;
+    onResizeMicroflow: string;
+    onResizeNanoflow: Nanoflow;
+    refreshInterval: number;
+    resizable: string;
+    customViews: CustomViews[];
+    dayFormat: string;
+    views: Views;
+    weekdayFormat: string;
+    timeGutterFormat: string;
+    monthHeaderFormat: string;
+    dayHeaderFormat: string;
+}
+
+type DataSource = "context" | "XPath" | "microflow" | "nanoflow";
+type OnClickEventOptions = "doNothing" | "callMicroflow" | "callNanoflow";
+type Views = "custom" | "standard";
+interface CalendarContainerState {
+    alertMessage: ReactChild;
+    events: CalendarEvent[];
+    eventColor: string;
+    startPosition: Date;
+    loading: boolean;
+}
+
+interface Nanoflow {
+    nanoflow: object[];
+    paramsSpec: { Progress: string };
+}
+
+interface ViewOptions {
+    month?: string;
+    week?: string;
+    work_week?: string;
+    day?: string;
+    agenda?: string;
+}
+
+export default class CalendarContainer extends Component<CalendarContainerProps, CalendarContainerState> {
```

Comment:
Class is to big
Split file create data source class handling retrieval, update and subscription

Reply:
Done